docs(v2): Docusaurus 2 beta blog post (#4756)
facebook/docusaurus: website-1.x/blog/2021-05-12-announcing-docusaurus-two-beta.md

author: Sébastien Lorber
authorTitle: Docusaurus maintainer
authorURL: https://sebastienlorber.com
authorImageURL: https://github.com/slorber.png
authorTwitter: sebastienlorber
tags: [release, beta]
image: /img/blog/2021-05-12-announcing-docusaurus-two-beta/social-card.png
---

After a lengthy alpha stage in order to ensure feature parity and quality, we are excited to officially release the first **Docusaurus 2 beta**.

With the announcement of this beta, the team is even more confident that Docusaurus 2 is **ready for mainstream adoption**!

![Docusaurus beta party](/img/blog/2021-05-12-announcing-docusaurus-two-beta/image_cropped.png)

<!--truncate-->

## Docusaurus adoption

**Don't fear the beta tag!**

Docusaurus 2 is widely adopted and growing fast:

[![Docusaurus growth](/img/blog/2021-05-12-announcing-docusaurus-two-beta/trend.png)](https://www.npmtrends.com/docusaurus-vs-@docusaurus/core)

To get a fuller understanding of the quality of current Docusaurus 2 sites, our new [showcase](https://docusaurus.io/showcase) page allows you to filter Docusaurus sites by features, so you may get inspired by real-world production sites with a similar use-case as yours!

> Don't miss our [favorite](https://docusaurus.io/showcase?tags=favorite) sites, they all stand out with something unique:

Reviewer: ```suggestion
Don't miss our [favorite](https://docusaurus.io/showcase?tags=favorite) sites; they all stand out with something unique:
```

We're now quite confident that the core features of Docusaurus 2 are stable enough to be in beta.

## What are the goals of the beta?

Now that Docusaurus 2 is stable and major feature complete, the goal of the beta is to inspire confidence in new users on the production-readiness of Docusaurus 2, migrate more remaining Docusaurus 1 users to version 2, and officially deprecate Docusaurus 1. We will, of course, continue to resolve any issues and bugs that may be discovered.

In addition, we will use the beta phase to **improve our theming system**.

We want to make it:

- **easier to implement a custom theme**, including for ourselves. We want to provide [multiple official themes](https://github.com/facebook/docusaurus/issues/3522) (including [Tailwind CSS](https://github.com/facebook/docusaurus/issues/2961)) for a long time.

- **safer to extend an existing theme**: it can be painful to upgrade a highly customized Docusaurus site, as customizations can conflict with internal changes. We need to make the theme public API surface more explicit, and make it clear what is safe to customize.

We will build a better **theming infrastructure** and refactor the classic theme to use it.

> If you don't customize your Docusaurus site much, you can consider Docusaurus stable.

Reviewer: ```suggestion
If you customize your site, you may find these planned improvements quite valuable.
```

In case you missed it, we recently shipped two major improvements:

- [Auto-generated sidebars](https://docusaurus.io/docs/sidebar#sidebar-item-autogenerated): no need to maintain a `sidebars.js` file anymore!
- [Webpack 5 / PostCSS 8](https://github.com/facebook/docusaurus/issues/4027): persistent caching significantly speeds up **rebuild time**!

## What's next?

Shipping the official 2.0 release!

To get there, we will continue to **fix bugs** and implement the **most wanted features**, including:

- [Sidebar category index pages](https://github.com/facebook/docusaurus/issues/2643)
- [Proper CSS-in-JS support](https://github.com/facebook/docusaurus/issues/3236)
- [Improve build time performance](https://github.com/facebook/docusaurus/issues/4765)
- [Extend Docusaurus plugins, CMS integration](https://github.com/facebook/docusaurus/issues/4138)
- [Fix trailing slashes and relative link issues](https://github.com/facebook/docusaurus/issues/3372)
- [Better compatibility with CommonMark](https://github.com/facebook/docusaurus/issues/3018)
- [Upgrade to MDX 2.0](https://github.com/facebook/docusaurus/issues/4029)

## Conclusion

This is an exciting time for Docusaurus.

> We read a [**lot** of positive feedback](https://twitter.com/sebastienlorber/timelines/1392048416872706049) about Docusaurus, and discover new sites online every single day.

Reviewer: ```suggestion
We are inspired by the [positive feedback](https://twitter.com/sebastienlorber/timelines/1392048416872706049) about Docusaurus, and discover new sites online every single day.
```

> We are so excited for this release, and hope that by moving from alpha to beta the community is assured on quality and stability, which, in turn, continues to increase the adoption of Docusaurus. And, for those that have been on the fence from migrating an existing site to Docusaurus 2, it is a great time to upgrade so that you are running on the latest infrastructure when we deprecate Docusaurus 1 at the end of this beta period.

Reviewer: ```suggestion
We are so excited for this release, and hope, by moving from alpha to beta, the community is assured on quality and stability, which, in turn, continues to increase the adoption of Docusaurus. And, for those that have been on the fence from migrating an existing Docusaurus site to Docusaurus 2, it is a great time to upgrade so that you are running on the latest infrastructure when we deprecate Docusaurus 1 at the end of this beta period.
```
